backend/src/main/java/ai/verta/modeldb/App.java

```diff
@@ -107,13 +108,16 @@ public static ApplicationContext getContext() {
 
   // Export all JMX metrics to Prometheus
   private static final String rules = "---\n" + "rules:\n" + "  - pattern: \".*\"";
+  private static final AtomicBoolean metricsInitialized = new AtomicBoolean(false);
```

Comment:
Should we use a mutex or is this never run in parallel?

Reply:
It's never run in parallel, and typically is only ever run exactly once.  In the tests the setup functionality calls this once per test during the setup phase, so access is always sequential.

Reply:
FWIW: We should refactor the the tests to avoid this, since it means that setup logic that is designed to run once is getting called more than once.  Normally it just doesn't matter, but the metrics collectors must be JVM-level singletons.